src/codegen/array_boundary.rs

```diff
@@ -0,0 +1,159 @@
+        }
+    } else if let ast::Expression::AllocDynamicArray(_, ty, size, _) = right {
+        let a = if cfg.array_lengths_temps.get(pos) != None {
+            // If there was a temp variable, reassign it to the new value
+            cfg.array_lengths_temps.remove(pos).unwrap()
+        } else {
+            //create a new temp variable
+            vartab.temp_name("array_size", ty)
+        };
+
+        if let ast::Expression::Variable(_, _, position) = **size {
+            let num = Expression::Variable(Loc::Codegen, Type::Uint(32), position);
+            cfg.add(
+                vartab,
+                Instr::Set {
+                    loc: Loc::Codegen,
+                    res: a,
+                    expr: num,
+                },
+            );
+
+            cfg.array_lengths_temps.insert(*pos, a);
+        }
+        // If size a uint and bits > 32
+        else if let ast::Expression::Trunc(_, _, ref index) = **size {
+            if let ast::Expression::Variable(_, _, index) = &**index {
+                // A number var holding array length
+                let num = Expression::Variable(Loc::Codegen, Type::Uint(32), *index);
+                let num_trunced = Expression::Trunc(Loc::Codegen, Type::Uint(32), Box::new(num));
+
+                cfg.add(
+                    vartab,
+                    Instr::Set {
+                        loc: Loc::Codegen,
+                        res: a,
+                        expr: num_trunced,
+                    },
+                );
+                cfg.array_lengths_temps.insert(*pos, a);
+            }
+        } else if let ast::Expression::ZeroExt(_, _, ref index) = **size {
+            if let ast::Expression::Variable(_, _, index) = &**index {
+                // A number var holding array length
+                let num = Expression::Variable(Loc::Codegen, Type::Uint(32), *index);
+                let num_trunced = Expression::ZeroExt(Loc::Codegen, Type::Uint(32), Box::new(num));
+
+                cfg.add(
+                    vartab,
+                    Instr::Set {
+                        loc: Loc::Codegen,
+                        res: a,
+                        expr: num_trunced,
+                    },
+                );
+                cfg.array_lengths_temps.insert(*pos, a);
+            }
+        }
+        // If the size is a number literal
+        else if let ast::Expression::NumberLiteral(_, _, size_of_array) = &**size {
+            //a number literal holding the array length
+            let num =
+                Expression::NumberLiteral(Loc::Codegen, Type::Uint(32), size_of_array.clone());
+            // let temp_res = vartab.temp_name("array_size", &Type::Uint(32));
+            cfg.add(
+                vartab,
+                Instr::Set {
+                    loc: Loc::Codegen,
+                    res: a,
+                    expr: num,
+                },
+            );
+            cfg.array_lengths_temps.insert(*pos, a);
+        }
```

Comment:
I don't know what this is trying to achieve.

Reply:
If re-allocation is done like this:
` uint[] arr = new uint256[](10);
arr = new `uint256[](20);`
I'm getting the Numberliteral inside AllocDynamicArray